nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,316 @@
+            .description("The Bearer Token provided by Twitter.")
+            .required(true)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor QUEUE_SIZE = new PropertyDescriptor.Builder()
+            .name("Queue Size")
+            .displayName("Queue Size")
+            .description("The size of the queue used to queue tweets")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("10000")
+            .build();
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .displayName("Batch Size")
+            .description("The maximum size of the number of tweets to be written to a single FlowFile." +
+                    "Will write less tweets if it there are not any tweets left in queue")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("10")
+            .build();
+    public static final PropertyDescriptor TWEET_FIELDS = new PropertyDescriptor.Builder()
+            .name("Tweet Fields")
+            .displayName("Tweet Fields")
+            .description("A comma-separated list of tweet fields to be returned as part of the tweet. Refer to " +
+                    "https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet " +
+                    "for proper usage. Possible field values are: " +
+                    "attachments, author_id, context_annotations, conversation_id, created_at, entities, geo, id, " +
+                    "in_reply_to_user_id, lang, non_public_metrics, organic_metrics, possibly_sensitive, promoted_metrics, " +
+                    "public_metrics, referenced_tweets, reply_settings, source, text, withheld")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor USER_FIELDS = new PropertyDescriptor.Builder()
+            .name("User Fields")
+            .displayName("User Fields")
+            .description("A comma-separated list of user fields to be returned as part of the tweet. Refer to " +
+                    "https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/user " +
+                    "for proper usage. Possible field values are: " +
+                    "created_at, description, entities, id, location, name, pinned_tweet_id, profile_image_url, " +
+                    "protected, public_metrics, url, username, verified, withheld")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor MEDIA_FIELDS = new PropertyDescriptor.Builder()
+            .name("Media Fields")
```

Comment:
All of the *_FIELDS should have defaults.

Reply:
Thanks for the comment. Although I have some thoughts. These are non-mandatory fields and serve as query parameters when the Twitter API class issues a GET request. A default value (and making this a required property) that makes sense for a user experience perspective would be 

- an empty string, or
- a comma separated list of every possible field that could be put in there

Any thoughts and recommendations? Thank you!

Reply:
I think the goal should be to return everything by default and then let the user pare it down as they want. If the empty string does the same thing as a comma-separated list, then that works.

Reply:
The Twitter [Fields](https://developer.twitter.com/en/docs/twitter-api/fields) documentation lists a large number of fields for each type, and the default value of empty string aligns with their REST API. Including all of the fields by default seems difficult to maintain, as the fields could change over time.

It seems best to provide links to the Twitter object model documentation, where each of the fields are documented, such as the following:

- [Tweets](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet)
- [Users](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/user)